atef/config.py

```diff
@@ -602,9 +606,10 @@ def walk_groups(
     ) -> Generator[PreparedGroup, None, None]:
         """Walk through the prepared groups."""
         for config in self.configs:
-            if isinstance(config, PreparedGroup):
+            if isinstance(config, get_args(AnyPreparedConfiguration)):
```

Comment:
For reference, this becomes:
```python
isinstance(config, (
    PreparedDeviceConfiguration,
    PreparedGroup,
    PreparedPVConfiguration,
    PreparedToolConfiguration
   )
 )
```

I'm a bit confused why `walk_groups` turned into this: it's yielding things that aren't just `PreparedGroup`
I guess the annotation needs updating or the function needs renaming?

Reply:
The annotation definitely needs updating, good point.  

My rationale here is that this method wants to walk dataclasses that hold **groups** of comparisons.  In my mind, these 4 subclasses of `PreparedConfiguration` are siblings in that sense.  

Pragmatically, this method wasn't used anywhere else and I just mutated it to suit my needs :)